pkg/process/runner/submitter_test.go

```diff
@@ -323,6 +327,44 @@ func Test_getRequestID(t *testing.T) {
 	assert.NotEqual(t, id1, id5)
 }
 
+func TestSubmitterHeartbeatProcess(t *testing.T) {
+	originalFlavor := flavor.GetFlavor()
+	defer flavor.SetFlavor(originalFlavor)
+	flavor.SetFlavor(flavor.ProcessAgent)
+
+	ctrl := gomock.NewController(t)
+	statsdClient := mockStatsd.NewMockClientInterface(ctrl)
+	statsdClient.EXPECT().Gauge("datadog.process.agent", float64(1), gomock.Any(), float64(1)).MinTimes(1)
+	processStatsd.Client = statsdClient
+
+	deps := newSubmitterDeps(t)
+	s, err := NewSubmitter(deps.Config, deps.Log, deps.Forwarders, testHostName)
+	assert.NoError(t, err)
+	s.heartbeatTime = 5 * time.Millisecond
+	s.Start()
+	time.Sleep(10 * time.Millisecond)
+	s.Stop()
+}
+
+func TestSubmitterHeartbeatCore(t *testing.T) {
+	originalFlavor := flavor.GetFlavor()
+	defer flavor.SetFlavor(originalFlavor)
+	flavor.SetFlavor(flavor.DefaultAgent)
+
+	ctrl := gomock.NewController(t)
+	statsdClient := mockStatsd.NewMockClientInterface(ctrl)
+	statsdClient.EXPECT().Gauge("datadog.process.agent", float64(1), gomock.Any(), float64(1)).Times(0)
+	processStatsd.Client = statsdClient
+
+	deps := newSubmitterDeps(t)
+	s, err := NewSubmitter(deps.Config, deps.Log, deps.Forwarders, testHostName)
+	assert.NoError(t, err)
+	s.heartbeatTime = 5 * time.Millisecond
+	s.Start()
+	time.Sleep(10 * time.Millisecond)
+	s.Stop()
```

Comment:
Please let me know if there alternatives to having sleeps here. I'm very open to changing this. I gave extra breathing room on the sleep to mitigate flakiness but not sure its enough.

Reply:
What if we inject a ticker into the `sendHeartbeat` function, so that you can create and access it during your test? That way we can just test `sendHeartbeat` instead of the submitter as a black box?

Reply:
I think thats better than the time in general. However, wouldn't going that route result in not testing the actual flavor check?

Reply:
We can just a the mock clock that way we can control the ticks.

Reply:
Ah yep, that slipped my mind. I will do that.

Reply:
Done